generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/NumericRestrictionsMergeOperation.java

```diff
@@ -1,27 +1,39 @@
 package com.scottlogic.deg.generator.fieldspecs;
 
+import com.google.inject.Inject;
 import com.scottlogic.deg.generator.constraints.atomic.IsOfTypeConstraint;
 import com.scottlogic.deg.generator.restrictions.*;
 
 import java.util.Optional;
 
 public class NumericRestrictionsMergeOperation implements RestrictionMergeOperation {
-    private static final NumericRestrictionsMerger numericRestrictionsMerger = new NumericRestrictionsMerger();
+    private final NumericRestrictionsMerger merger;
+
+    @Inject
+    public NumericRestrictionsMergeOperation(NumericRestrictionsMerger merger) {
+        this.merger = merger;
+    }
 
     @Override
     public Optional<FieldSpec> applyMergeOperation(FieldSpec left, FieldSpec right, FieldSpec merging) {
-        MergeResult<NumericRestrictions> mergeResult = numericRestrictionsMerger.merge(
+        MergeResult<NumericRestrictions> mergeResult = merger.merge(
             left.getNumericRestrictions(), right.getNumericRestrictions());
 
         if (!mergeResult.successful) {
-            return Optional.empty();
+            TypeRestrictions typeRestrictions = merging.getTypeRestrictions();
+            if (typeRestrictions == null){
+                typeRestrictions = DataTypeRestrictions.ALL_TYPES_PERMITTED;
+            }
+
+            return Optional.of(merging
+                .withTypeRestrictions(
+                    typeRestrictions.except(IsOfTypeConstraint.Types.NUMERIC),
+                    left.getFieldSpecSource().combine(right.getFieldSpecSource())));
         }
 
         NumericRestrictions numberRestrictions = mergeResult.restrictions;
         if (numberRestrictions == null) {
-            return Optional.of(merging.withNumericRestrictions(
-                null,
-                FieldSpecSource.Empty));
+            return Optional.of(merging);
```

Comment:
same as above comment

Reply:
because there are no dateTime restrictions to add to the `FieldSpec`